client/components/list.jsx

```diff
@@ -6,34 +6,43 @@ export default class List extends React.Component {
     this.state = {
       post: []
     };
+    this.toDetail = this.toDetail.bind(this);
   }
 
   componentDidMount() {
-    fetch('api/post')
+    fetch('api/main')
       .then(res => res.json())
-      .then(data =>
-        this.setState({ post: data }));
+      .then(data => {
+        this.setState({ post: data });
+      });
+  }
+
+  toDetail(event) {
+    const targetElement = event.target.closest('a');
+    const targetId = Number(targetElement.id);
+    window.location.hash = `post/${targetId}`;
   }
```

Comment:
Instead of listening for clicks, make the `href` of the `<a>` element the target hash with a query string.